src/compiler/lsp/multiplex_stdout.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2022 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+#include "multiplex_stdout.h"
+
+#include <stdio.h>
+
+namespace toit {
+namespace compiler {
+
+static void checked_fwrite(const void* data,
+                           int size,
+                           const char* error_message = "Couldn't write data") {
+  int written_bytes = fwrite(data, 1, size, stdout);
+  if (written_bytes != size) FATAL(error_message);
+}
+
+void LspWriterMultiplexStdout::printf(const char* format, va_list& arguments) {
+  va_list copy;
+  va_copy(copy, arguments);
+  int32 needed_bytes = static_cast<int32>(vsnprintf(null, 0, format, arguments));
+  checked_fwrite(&needed_bytes, sizeof(needed_bytes));
+  int written_bytes = vprintf(format, copy);
+  if (written_bytes < 0) {
+    perror("multi-printf");
```

Comment:
Why multi?

Reply:
I typically use something like perror("LspWriterMultiplexStdout/vprintf") to indicate which function failed and where the call was from-ish.

Reply:
Used `LspWriterMultiplexStdout::printf/vprintf`